bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -122,15 +122,16 @@ public static synchronized void runCleanupJobs() {
    *
    * @param job The {@code Job} to keep track of.
    */
-  public JobInfo waitForJob(Job job) {
+  public JobInfo waitForJob(Job job, long jobTimeoutInMinutes) {
```

Comment:
Should we instead make it a member of BigQueryClient? (Like `materializationDataset`)